src/shared/utils/prismLanguageMapper.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Maps file extension names to prism supported languages.
+ * The full list can be found in the prism-react-renderer package:
+ * https://github.com/FormidableLabs/prism-react-renderer/blob/v1.3.5/src/vendor/prism/includeLangs.js
+ */
+import * as Sentry from '@sentry/react'
+import { type Language } from 'prism-react-renderer'
+
+const prismSupportedLanguages = new Map<string, Language>([
+  ['html', 'markup'],
+  ['xml', 'markup'],
+  ['svg', 'markup'],
+  ['mathml', 'markup'],
+  ['ssml', 'markup'],
+  ['atom', 'markup'],
+  ['rss', 'markup'],
+  ['sh', 'bash'],
+  ['c', 'c'],
+  ['h', 'clike'],
+  ['cc', 'cpp'],
+  ['cpp', 'cpp'],
+  ['css', 'css'],
+  ['js', 'javascript'],
+  ['cjs', 'javascript'],
+  ['mjs', 'javascript'],
+  ['jsx', 'jsx'],
+  ['go', 'go'],
+  ['gql', 'graphql'],
+  ['graphql', 'graphql'],
+  ['json', 'json'],
+  ['less', 'less'],
+  ['objc', 'objectivec'],
+  ['ocaml', 'ocaml'],
+  ['py', 'python'],
+  ['reason', 'reason'],
+  ['sass', 'sass'],
+  ['scss', 'scss'],
+  ['sql', 'sql'],
+  ['ts', 'typescript'],
+  ['tsx', 'tsx'],
+  ['wasm', 'wasm'],
+  ['yaml', 'yaml'],
+])
+
+const DEFAULT_LANGUAGE_TYPE: Language = 'markup'
+
+export function prismLanguageMapper(fileName: string): Language {
+  const fileExtension = fileName.split('.').pop() ?? ''
```

Comment:
Thoughts on if there is no `.` then we should just render it as markup?

Reply:
There could be multiple ways, one is to just not render any tokens since we cant tell what file type it is, the other could be to set a value like markup, but then also allow users to select a different language (they'll know best what the contents could be)

Reply:
Hrrm alrighty, I wonder how big of problem this may be. So just some background information, Codecov will only display files in the UI that we have coverage for, so if you have a file like `.prettierrc` unless you somehow write tests for that, it won't show up in the UI. With that in mind I think we should rollout what we have now, and if things blow up then we can tackle them there. Adding in a selector for selecting languages seems ... a little complex to add into this PR.

Reply:
No, I def wouldnt add it to this PR, but it might make sense to progressively enhance this later on. Defaulting to markup seems like a good default for now